### PR TITLE
Remove old plan generation APIs

### PR DESCRIPTION
**Pull request by jingtang10**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

**Description**
Delete deprecated old plan generation APIs in favour of new ones with optional parameters. This makes updating to the latest version of CQL stack easier.

Deprecated APIs to delete:
```
FhirOperator.generateCarePlan(String,String): IBaseResource
FhirOperator.generateCarePlan(String,String,String): IBaseResource
FhirOperator.generateCarePlan(CanonicalType,String,String? = null,String? = null,String? = null,IBaseDatatype? = null,IBaseDatatype? = null,IBaseDatatype? = null,IBaseDatatype? = null,IBaseDatatype? = null,IBaseParameters? = null,Boolean? = null,IBaseBundle? = null,IBaseParameters? = null): IBaseResource
```

All merged into the same API.

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: Code health

**Screenshots (if applicable)**

**Checklist**

- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://google.github.io/android-fhir/contrib/) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

File: workflow/src/main/java/com/google/android/fhir/workflow/FhirOperator.kt [hunk 1]
```diff
@@ -154,93 +154,11 @@ internal constructor(
...
-      /* bundle = */ bundle,
-      /* prefetchData = */ prefetchData,
-      libraryProcessor,
-    ) as IBaseResource
-  }
-
   @WorkerThread
   fun generateCarePlan(
-    planDefinition: PlanDefinition,
+    id: String? = null,
```
Comment (MJ1998) [inline workflow/src/main/java/com/google/android/fhir/workflow/FhirOperator.kt:159]: NIT
```suggestion
    planDefinitionId: String? = null,
```

**Review by MJ1998 — COMMENTED** [review]
By "deprecate" do you mean remove ? The deprecated APIs are being removed. Is that intended ?

File: workflow/src/main/java/com/google/android/fhir/workflow/FhirOperator.kt (hunk 1 above)
Comment (jingtang10) [reply workflow/src/main/java/com/google/android/fhir/workflow/FhirOperator.kt]: done. also renamed plan defintion canonical

**Review by aditya-07 — APPROVED** [review]
lgtm

**Pull request merged**